testharness.js tests do not work when loaded in cross-origin iframes

kkinnunen-apple opened the pull request:
#### 2fa5620a0732536cd1601232eb89ab07654ef4cf
<pre>
testharness.js tests do not work when loaded in cross-origin iframes
<a href="https://bugs.webkit.org/show_bug.cgi?id=280048">https://bugs.webkit.org/show_bug.cgi?id=280048</a>
<a href="https://rdar.apple.com/136347990">rdar://136347990</a>

Reviewed by Alex Christensen.

Add a url parameter inCrossOriginFrame=true to the iframe src that
is used for the WebKitTestRunner cross-origin frame test mode.

The WebKit specific testharnessreport.js contains a filter that no-ops
test completion reports if the test is loaded from an iframe. Avoid
this filter if the document is indeed the main document, loaded from
the cross-origin frame.

Fixes testharness.js test timeouts with run-webkit-tests --site-isolation.

* LayoutTests/resources/testharnessreport.js:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::loadTestInCrossOriginIframe):

Canonical link: <a href="https://commits.webkit.org/284092@main">https://commits.webkit.org/284092@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b60999f75ca7335eb86dfb634a2a65e93d882c9e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67993 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/47385 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20652 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/72052 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/19138 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/70110 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/55181 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18954 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54344 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/12758 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/71060 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43398 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58760 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34812 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/40067 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/16177 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/17495 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62041 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16518 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73748 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11964 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15802 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61805 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/12002 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58836 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61837 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15175 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9719 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3331 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/43185 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/44259 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/45454 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/44001 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on LayoutTests/resources/testharnessreport.js:
```diff
@@ -61,6 +61,7 @@ if (self.testRunner) {
 
     // window.opener is a configurable property, so store it before we run anything.
     const orig_opener = window.opener;
+    const isInCrossOriginFrame = window.location.href.indexOf("inCrossOriginFrame=true") >= 0;
```
Could we check href.searchParams.has("inCrossOriginFrame", "true")?
Also, could we make this even more explicit "inCrossOriginFrameForSiteIsolationTesting" or something?

kkinnunen-apple committed:
testharness.js tests do not work when loaded in cross-origin iframes
https://bugs.webkit.org/show_bug.cgi?id=280048
rdar://136347990

Reviewed by Alex Christensen.

Add a url parameter inCrossOriginFrame=true to the iframe src that
is used for the WebKitTestRunner cross-origin frame test mode.

The WebKit specific testharnessreport.js contains a filter that no-ops
test completion reports if the test is loaded from an iframe. Avoid
this filter if the document is indeed the main document, loaded from
the cross-origin frame.

Fixes testharness.js test timeouts with run-webkit-tests --site-isolation.

* LayoutTests/resources/testharnessreport.js:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::loadTestInCrossOriginIframe):

Canonical link: https://commits.webkit.org/284092@main

gsnedders commented on Tools/WebKitTestRunner/TestInvocation.cpp:
```diff
@@ -157,7 +157,7 @@ void TestInvocation::loadTestInCrossOriginIframe()
         "    testRunner.dumpChildFramesAsText()"
         "</script>"
-        "<iframe src=\""_s, m_urlString.utf8().span(), "\" style=\"position:absolute; top:0; left:0; width:100%; height:100%; border:0\">"_s));
+        "<iframe src=\""_s, m_urlString.utf8().span(), "?inCrossOriginFrame=true\" style=\"position:absolute; top:0; left:0; width:100%; height:100%; border:0\">"_s));
```
Note there are tests loaded which already have query strings and/or fragments, where this won't work.
